src/main.cpp

```diff
@@ -208,83 +208,95 @@ bool validate_format_extension(std::wstring ext_w) {
 #endif
 
 
-#ifdef HAVE_OPENCV_3_X
+#if defined(HAVE_OPENCV) && defined(HAVE_OPENCV_3_X)
 	#define w2xHaveImageWriter(x) cvHaveImageWriter(x)
 #else
 	#define w2xHaveImageWriter(x) cv::haveImageWriter(x)
 #endif
 	
-//check for opencv formats
-void check_opencv_formats()
+//check for supported formats
+void check_supported_formats()
 {
+	#ifndef HAVE_OPENCV
+		// Only default formats are supported
+		return;
+	#else
 	// Portable Network Graphics
 	if (!w2xHaveImageWriter(".png"))
 	{
-		opencv_formats["PNG"] = false;
+		supported_formats["PNG"] = false;
 	}
 	
 	// JPEG Files
 	if (!w2xHaveImageWriter(".jpg"))
 	{
-		opencv_formats["JPEG"] = false;
-		opencv_formats["JPG"] = false;
-		opencv_formats["JPE"] = false;
+		supported_formats["JPEG"] = false;
+		supported_formats["JPG"] = false;
+		supported_formats["JPE"] = false;
 	}
 	
 	/* 
 	Disabled due to vulnerabilities in Jasper codec, see: https://github.com/opencv/opencv/issues/14058
 	// JPEG 2000 Files
 	if (!w2xHaveImageWriter(".jp2"))
 	{
-		opencv_formats["JP2"] = false;
+		supported_formats["JP2"] = false;
 	}
 	*/
 	
 	// WebP Files
 	if (!w2xHaveImageWriter(".webp"))
 	{
-		opencv_formats["WEBP"] = false;
+		supported_formats["WEBP"] = false;
 	}
 	
 	// TIFF Files
 	if (!w2xHaveImageWriter(".tif"))
 	{
-		opencv_formats["TIF"] = false;
-		opencv_formats["TIFF"] = false;
+		supported_formats["TIF"] = false;
+		supported_formats["TIFF"] = false;
 	}
 	
 	// OpenEXR Image Files
 	if (!w2xHaveImageWriter(".exr"))
 	{
-		opencv_formats["EXR"] = false;
+		supported_formats["EXR"] = false;
 	}
 
 	/* These formats are always available.
 	// Windows Bitmaps (Always Supported)
-	opencv_formats["BMP"] = true;
-	opencv_formats["DIB"] = true;
+	supported_formats["BMP"] = true;
+	supported_formats["DIB"] = true;
 	
 	// Portable Image Format (Always Supported)
-	opencv_formats["PBM"] = true;
-	opencv_formats["PGM"] = true;
-	opencv_formats["PPM"] = true;
-	opencv_formats["PXM"] = true;
-	opencv_formats["PNM"] = true;
+	supported_formats["PBM"] = true;
+	supported_formats["PGM"] = true;
+	supported_formats["PPM"] = true;
+	supported_formats["PXM"] = true;
+	supported_formats["PNM"] = true;
 	
 	// Sun Rasters (Always Supported)
-	opencv_formats["SR"] = true;
-	opencv_formats["RAS"] = true;
+	supported_formats["SR"] = true;
+	supported_formats["RAS"] = true;
 	
 	// Radiance HDR (Always Supported)
-	opencv_formats["HDR"] = true;
-	opencv_formats["PIC"] = true;
+	supported_formats["HDR"] = true;
+	supported_formats["PIC"] = true;
 	*/
+	#endif
 }
 
-void debug_show_opencv_formats()
+void display_supported_formats()
 {
-	std::cout << "This is a list of supported formats, it depends on which formats opencv has been built with." << std::endl ;
-	for (auto const& x : opencv_formats)
+	std::cout
+	#ifdef HAVE_OPENCV
+			 << " [With OpenCV] This list depends on which formats opencv has been built with."
+	#else
+			 << " [Without OpenCV] (recompile to enable), Only the default formats can be used."
```

Comment:
> nable), Only the default
I recommend either replacing the comma (`,`) with a stop (`.`) or make the `O` in "Only" small.